src/astro/sql/operators/agnostic_stats_check.py

```diff
@@ -0,0 +1,392 @@
+from distutils import log as logger
+from os import name, stat
+from typing import Dict, List, Set
+
+from airflow.hooks.base import BaseHook
+from airflow.providers.postgres.hooks.postgres import PostgresHook
+from psycopg2 import sql
+from sqlalchemy.sql.expression import table
+from sqlalchemy.sql.functions import Function
+from sqlalchemy.sql.schema import Table
+
+from astro.sql.operators.sql_decorator import SqlDecoratoratedOperator
+from astro.sql.table import Table
+from astro.utils.snowflake_merge_func import (
+    is_valid_snow_identifier,
+    is_valid_snow_identifiers,
+)
+
+
+class OutlierCheck:
+    def __init__(
+        self,
+        name: str,
+        columns_map: map,
+        accepted_std_div: int,
+        threshold: float,
+    ) -> None:
+        self.name = name
+        self.columns_map = columns_map
+        self.accepted_std_div = accepted_std_div
+        self.threshold = threshold
+
+
+class ChecksHandler:
```

Comment:
Placing this into its own class is a really great call. Will make it easier if we decide to change how we handle checks in the future.